src/shared/components/Menu/NavDrawer.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import RoutedComponent, { Route } from './RoutedComponent';
+import SideMenu from './SideMenu';
+import './NavDrawer.less';
+
+interface NavDrawerComponentProps {
+  render: (
+    visible: boolean,
+    toggleVisibility: () => void
+  ) => React.ReactElement<any>;
+  routes: Route[];
+  defaultVisibility?: boolean;
+}
+
+export const NavDrawer: React.FunctionComponent<
+  NavDrawerComponentProps
+> = props => {
+  const { render, defaultVisibility, routes } = props;
+
+  const [visible, setVisible] = React.useState(
+    typeof defaultVisibility !== 'undefined' ? defaultVisibility : true
```

Comment:
Since `visible` has to be a Boolean, why not write the more explicit `typeof defaultVisibility === boolean`?

Reply:
hmm, I would keep it simple and just do:
```suggestion
    defaultVisibility || true
```